pkg/apis/core/types.go

```diff
@@ -3510,7 +3510,8 @@ type ServiceSpec struct {
 
 	// publishNotReadyAddresses, when set to true, indicates that DNS implementations
 	// must publish the notReadyAddresses of subsets for the Endpoints associated with
-	// the Service. The default value is false.
+	// the Service. This causes all endpoints derived from the Service's selector to be marked
+	// ready, regardless of the backend Pod's status. The default value is false.
 	// The primary use case for setting this field is to use a StatefulSet's Headless Service
 	// to propagate SRV records for its Pods without respect to their readiness for purpose
 	// of peer discovery.
```

Comment:
Suggest putting the selector-aspect after the "primary use case" sentence, since not all headless services have selectors (https://kubernetes.io/docs/concepts/services-networking/service/#with-selectors).

```suggestion
	// the Service. The default value is false.
	// The primary use case for setting this field is to use a StatefulSet's Headless Service
	// to propagate SRV records for its Pods without respect to their readiness for purpose
	// of peer discovery.
	// If this is set to true on a service with a selector, all endpoints derived from the Service's
	// selector will be marked as ready, regardless of the backend Pod's status.
```

Reply:
Makes sense, done